### PR TITLE
Basic v1 account creation API

### PR DESCRIPTION
**Pull request by gfodor**
Adds a basic API for creating accounts and looking up accounts by email, accessible to administrator accounts only.

The create API attempts to be a first stab at a idiomatic public API. It has a few characteristics:

- Allows single or multiple record posting, with multiple records being handled via an HTTP 207 response
- Uses JSON schema for validation
- Has fairly robust error handling (at least compared to what we've had in the past :))

Note the lookup API uses a POST because otherwise emails would risk being logged somewhere.

### CODE REVIEW

File: lib/ret_web/controllers/api/v1/account_search_controller.ex [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  alias Ret.{Account}
+  alias RetWeb.Api.V1.{AccountView}
+
+  # Account lookup, is a POST because lookup contains sensitive information
+  def create(conn, %{"email" => email}) do
+    with %Account{} = account <- Account.account_for_email(email) do
+      record = Phoenix.View.render(AccountView, "show.json", account: account)
+      conn |> send_resp(200, %{records: [record]} |> Poison.encode!())
+    else
+      _ -> conn |> put_status(:not_found) |> halt()
```
Comment (brianpeiris) [inline lib/ret_web/controllers/api/v1/account_search_controller.ex]: Would it make sense to respond with an `{errors: ...}` json in this case as well? It's generally nicer for an API to always return json if it normally returns json.

File: lib/ret_web/controllers/api/v1/account_controller.ex [hunk 2]
```diff
@@ -0,0 +1,34 @@
+defmodule RetWeb.Api.V1.AccountController do
+  use RetWeb, :controller
+  import RetWeb.ApiHelpers
+
+  alias Ret.{Account}
+  alias RetWeb.Api.V1.{AccountView}
+
+  # TODO move to a file
```
Comment (brianpeiris) [inline lib/ret_web/controllers/api/v1/account_controller.ex]: Remove/resolve TODO

File: lib/ret_web/api_helpers.ex [hunk 3]
```diff
@@ -0,0 +1,86 @@
+defmodule RetWeb.ApiHelpers do
+  import Plug.Conn
+
+  # In an API create routine, call this with a handler.
+  # The handler will get passed the (validated) record and source and should return either:
+  #   { :ok, { http_status, result } }
+  #   { :error, [ { code, details, source } ] }
+  # TODO dialyzer this
```
[the remark this answers is not in the document]
Comment (gfodor) [reply lib/ret_web/api_helpers.ex]: ah that was more for longer term once we start caring about dialyzer. will take out for now

File: lib/ret_web/controllers/api/v1/account_search_controller.ex (hunk 1 above)
Comment (gfodor) [reply lib/ret_web/controllers/api/v1/account_search_controller.ex]: i think for the 404's this seems right. we're already `halt()`ing for 401s in other places, but i'll update the 404 paths.

File: lib/ret_web/controllers/api/v1/account_controller.ex (hunk 2 above)
Comment (gfodor) [reply lib/ret_web/controllers/api/v1/account_controller.ex]: this was also longer term (ie when we want to have swagger docs, etc). will remove

**Pull request closed without merging**